lib/ocpp/v201/charge_point.cpp

```diff
@@ -1134,6 +1134,15 @@ void ChargePoint::message_callback(const std::string& message) {
         this->security->security_event_notification_req(CiString<50>(security_event), CiString<255>(message), true,
                                                         utils::is_critical(security_event));
         return;
+    } catch (const StringConversionException& e) {
+        this->logging->central_system("Unknown", message);
+        EVLOG_error << "JSON exception during reception of message: " << e.what();
+        this->message_dispatcher->dispatch_call_error(
+            CallError(MessageId("-1"), "RpcFrameworkError", e.what(), json({})));
+        const auto& security_event = ocpp::security_events::INVALIDMESSAGES;
+        this->security->security_event_notification_req(CiString<50>(security_event), CiString<255>(message), true,
```

Comment:
This can still throw if `message` > 255 characters or `security_event` > 50 characters.

It might be an option to truncate CiStrings by default inside the class in case the number of characters exceeds the specified limit instead of throwing the exception. I dont have a complete overview but I think it can happen at quite a few places

Reply:
I've added an argument that can optionally truncate if we want. I don't think truncating by default is an option because that might lead to strange behavior as well. We would silently accept incorrect messages from the BO for example.